timeseries/src/autogluon/timeseries/trainer/abstract_trainer.py

```diff
@@ -35,130 +35,7 @@
-        load_path = os.path.join(path, cls.trainer_file_name)
-        if not reset_paths:
-            return load_pkl.load(path=load_path)
-        else:
-            obj = load_pkl.load(path=load_path)
-            obj.set_contexts(path)
-            obj.reset_paths = reset_paths
-            return obj
-
-    def load_model(
-        self,
-        model_name: Union[str, AbstractModel],
-        path: Optional[str] = None,
-        model_type: Optional[Type[AbstractModel]] = None,
-    ) -> AbstractTimeSeriesModel:
-        if isinstance(model_name, AbstractModel):
-            assert isinstance(model_name, AbstractTimeSeriesModel)
-            return model_name
-        if model_name in self.models.keys():
-            return self.models[model_name]
-
-        path_ = path if path is not None else self.get_model_attribute(model=model_name, attribute="path")
-        type_ = model_type if model_type is not None else self.get_model_attribute(model=model_name, attribute="type")
-
-        return type_.load(path=os.path.join(self.path, path_), reset_paths=self.reset_paths)
-
-    def get_models_info(self, models: Optional[List[str | AbstractModel]] = None) -> Dict[str, Any]:
-        models_ = models if models is not None else self.get_model_names()
-        model_info_dict = dict()
-        for model in models_:
-            if isinstance(model, str):
-                if model in self.models.keys():
-                    model = self.models[model]
-            if isinstance(model, str):
-                model_type = self.get_model_attribute(model=model, attribute="type")
-                model_path = os.path.join(self.path, self.get_model_attribute(model=model, attribute="path"))
-                model_info_dict[model] = model_type.load_info(path=model_path)
-            else:
-                model_info_dict[model.name] = model.get_info()
-        return model_info_dict
-
-    def get_info(self, include_model_info: bool = False, **kwargs) -> Dict[str, Any]:
-        num_models_trained = len(self.get_model_names())
-        if self.model_best is not None:
-            best_model = self.model_best
-        else:
-            try:
-                best_model = self.get_model_best()
-            except AssertionError:
-                best_model = None
-        if best_model is not None:
-            best_model_score_val = self.get_model_attribute(model=best_model, attribute="val_score")
-        else:
-            best_model_score_val = None
-
-        info = {
-            "best_model": best_model,
-            "best_model_score_val": best_model_score_val,
-            "num_models_trained": num_models_trained,
-        }
-
-        if include_model_info:
-            info["model_info"] = self.get_models_info()
-
-        return info
-
-
-class AbstractTimeSeriesTrainer(SimpleAbstractTrainer):
+class AbstractTimeSeriesTrainer(AbstractTrainer[AbstractTimeSeriesModel]):
```

Comment:
Why do we have a separate class for `AutoTrainer(AbstractTabularTrainer)` and `AutoTimeSeriesTrainer(AbstractTimeSeriesTrainer)`? Should these be merged into respective single classes as well?

Reply:
Right now this class hierarchy for timeseries is there for consistency with tabular. However I agree we can merge into a single `TimeSeriesTrainer` for timeseries. Wdyt?

Reply:
i.e., now I think we have everything in place to do that.

Reply:
Let's merge them, it's something that bothered me for a while.